### PR TITLE
Use a different mqtt client id for each connection

### PR DESCRIPTION
**Pull request by feclare**
There could only be one connection to a mqtt broker with the same id. Adding some randomness to the id

### CODE REVIEW

File: distro/mqtt.go [hunk 1]
```diff
@@ -26,25 +27,27 @@ const topic = "EdgeX"
 
 // NewMqttSender - create new mqtt sender
 func NewMqttSender(addr export.Addressable) Sender {
+	clientStr := clientID + "_" + strconv.FormatInt(rand.Int63(), 16)
```
Comment (drasko) [inline distro/mqtt.go]: Not mandatory, but proposition: add a comment why we are using `strconv.FormatInt(rand.Int63(), 16)` to form this string. I do not know intricacies, and I can not conclude from just reading the code.

Comment (feclare) [reply distro/mqtt.go]: I was just making sure there are not using the same mqtt id for several connections. The id is just a string, without any requirement.

Anyway, I forgot to set the seed for the RNG. I will commit it to this PR.

Comment (drasko) [reply distro/mqtt.go]: If so - please use `uuid()` generator maybe?

Comment (drasko) [reply distro/mqtt.go]: Something like this: https://github.com/satori/go.uuid or this: https://github.com/google/uuid

Comment (feclare) [reply distro/mqtt.go]: Yes, it is a good idea. Any lib recommendation? 

I'm thinking on using https://github.com/satori/go.uuid after a quick search.

Comment (feclare) [reply distro/mqtt.go]: ok, will try https://github.com/satori/go.uuid

File: distro/mqtt.go [hunk 2]
```diff
@@ -26,25 +27,29 @@ const topic = "EdgeX"
 
 // NewMqttSender - create new mqtt sender
 func NewMqttSender(addr export.Addressable) Sender {
+	// ClientId needs to be unique for each connection, otherwise the broker can
+	// close previous connection with the same id
+	clientStr := clientID + "_" + uuid.NewV4().String()
```
Comment (drasko) [inline distro/mqtt.go]: How is this `clientID` generated and can we use UUID here? I know that on connection to MQTT broker you can choose your clientID, but is it maybe something that here comes from registration services?

Comment (feclare) [reply distro/mqtt.go]: Yes, you are right. We don't need to generate the clientid, it is the Publisher field in the addressable.
Will fix it.

Comment (feclare) [reply distro/mqtt.go]: Done

**Comment by drasko** [comment]
Merged, thanks.

**Pull request merged**